src/pages/ServiceDetails/ServiceMetrics.tsx

```diff
@@ -1,18 +1,134 @@
 import * as React from 'react';
+import axios from 'axios';
 import ServiceId from '../../types/ServiceId';
+import { MetricHistogram, MetricValue } from '../../types/Metrics';
 
-class ServiceMetrics extends React.Component<ServiceId> {
+type ServiceMetricsState = {
+  range: string;
+  requestCountIn?: MetricValue;
+  requestCountOut?: MetricValue;
+  requestSizeIn?: MetricHistogram;
+  requestSizeOut?: MetricHistogram;
+  requestDurationIn?: MetricHistogram;
+  requestDurationOut?: MetricHistogram;
+  responseSizeIn?: MetricHistogram;
+  responseSizeOut?: MetricHistogram;
+  healthyReplicas?: MetricValue;
+  totalReplicas?: MetricValue;
+};
+
+class ServiceMetrics extends React.Component<ServiceId, ServiceMetricsState> {
   constructor(props: ServiceId) {
     super(props);
+    this.state = {
+      range: '5m'
+    };
+    this.onRangeChanged = this.onRangeChanged.bind(this);
+    this.fetchMetrics();
+  }
+
+  onRangeChanged(event: React.FormEvent<HTMLSelectElement>) {
+    this.setState({ range: event.currentTarget.value }, () => {
+      this.fetchMetrics();
+    });
   }
 
   render() {
     return (
       <div>
-        == SERVICE METRICS ({this.props.namespace} / {this.props.service}) ==
+        <h1>
+          == SERVICE METRICS ({this.props.namespace} / {this.props.service}) ==
+        </h1>
+        Range:
+        <select value={this.state.range} onChange={this.onRangeChanged}>
+          <option value="1m">1 minute</option>
```

Comment:
At some point we should probably make this data driven and not hard coded.

Reply:
@mtho11 I've read your comment too late. You're right, I must bring changes on bother core&ui